### PR TITLE
SIL: Bridge 'any Sendable' via the id-as-Any path

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes rdar://problem/107264240.

### CODE REVIEW

File: include/swift/AST/Types.h [hunk 1]
```diff
@@ -601,6 +601,9 @@ class alignas(1 << TypeAlignInBits) TypeBase
   /// Is this the 'Any' type?
   bool isAny();
 
+  /// Is this the 'any Sendable' type?
+  bool isAnySendable();
```
Comment (DougGregor) [inline include/swift/AST/Types.h:605]: Can we generalize this to be "`Any` ignoring marker protocols"?

Comment (slavapestov) [reply include/swift/AST/Types.h]: Alright, fine.

**Review by DougGregor — APPROVED** [review]

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**